sympy/physics/continuum_mechanics/beam.py

```diff
@@ -491,6 +528,108 @@ def apply_support(self, loc, type="fixed"):
         else:
             return reaction_load, reaction_moment
 
+    def apply_hinge(self, loc, type="rotation"):
```

Comment:
How many hinge types do you imagine there being? If only these two, why not make two separate methods. Then you wouldn't return different things depending on the kwarg.

e.g. `apply_rotation_hinge()` and `apply_sliding_hinge()`

Reply:
I agree with the main reason that sliding hinges are not very common

Reply:
These are the only two types I imagine being there. I will make two separate methods and then push again.